python/lsst/source/injection/consolidate_injected_catalogs.py

```diff
@@ -0,0 +1,283 @@
+        "brightness profiles defined by GalSim.",
+        default="source_type",
+    )
+
+
+class ConsolidateInjectedCatalogsTask(PipelineTask):
+    """Class for combining all tables in a collection of input catalogs
+    into one table.
+    """
+
+    _DefaultName = "consolidateInjectedCatalogsTask"
+    ConfigClass = ConsolidateInjectedCatalogsConfig
+
+    def runQuantum(self, butler_quantum_context, input_refs, output_refs):
+        inputs = butler_quantum_context.get(input_refs)
+        catalog_dict, tract = self._get_catalogs(inputs, input_refs)
+        outputs = self.run(catalog_dict, inputs["skyMap"], tract)
+        butler_quantum_context.put(outputs, output_refs)
+
+    def _get_catalogs(
+        self,
+        inputs: dict,
+        input_refs: InputQuantizedConnection,
+    ) -> tuple[dict, int]:
+        """Organize input catalogs into a dictionary with photometry band
+        keys.
+        """
+        catalog_dict: dict[str, list] = {}
+        tracts = set()
+        for ref, catalog in zip(input_refs.input_catalogs, inputs["input_catalogs"]):
+            band = ref.dataId.band.name
+            if band not in catalog_dict:
+                catalog_dict[band] = []
+            # load the patch number to check for patch overlap duplicates later
+            catalog["patch"] = ref.dataId.patch.id
+            catalog_dict[band].append(catalog)
+            tracts.add(ref.dataId.tract.id)
+        # vstack all the catalogs for each band
+        for band, catalog_list in catalog_dict.items():
+            catalog_dict[band] = vstack(catalog_list)
+        if len(tracts) != 1:
+            raise RuntimeError(f"len({tracts=}) != 1")
+        return (catalog_dict, list(tracts)[0])
+
+    def _remove_duplicates(
+        self,
+        catalog: Table,
+        tractInfo,
+    ) -> Table:
+        """Remove tract and patch overlap duplicates."""
+        self.config = cast(ConsolidateInjectedCatalogsConfig, self.config)
+        # TODO: add isPatchInner column
+        duplicates = []
+        for ind, row in enumerate(catalog):
+            tractInner = tractInfo.getInnerSkyPolygon()
+            # check if the source is within inner tract bounds
+            if tractInner.contains(np.deg2rad(row[self.config.col_ra]), np.deg2rad(row[self.config.col_dec])):
+                spherePoint = SpherePoint(
+                    row[self.config.col_ra] * degrees, row[self.config.col_dec] * degrees
+                )
+                # get the patch number from the source's ra,dec
+                patchInfo = tractInfo.findPatch(spherePoint)
+                # check against the patch column
+                if row["patch"] != patchInfo.sequential_index:
+                    duplicates.append(ind)
+            # if the source is not within inner tract bounds, remove it
+            else:
+                duplicates.append(ind)
+        catalog.remove_rows(duplicates)
+        return catalog
```

Comment:
If there is only one tract does this remove non duplicated sources?

Reply:
That's a great question...

Reply:
After doing some searching on slack it looks like I should remove this since each output catalog only covers a single tract. Good catch.

Reply:
I think we should keep the duplicates before and at the matching stage, and then people can remove them with isPrimary later on